### PR TITLE
Merge fixes and missed features

### PR DESCRIPTION
**Pull request by ktkrg**
*Description of changes:*
This commit merges some of the fixes and features that should have been in the split version of PA.

Features and fixes introduced in this PR:
1. Allow performance analyzer to be en(dis)abled through a cluster setting across the cluster.
1. Allow logging to be controllable through the cluster setting
1. Capture a node's role along with the host address and the node name
1. Checkstyle compliance

Some issues that are still not addressed in this PR:
1. Update build scripts to start the agent from the reader location instead of the plugin location
1. Remove pa_config, pa_bin and other folders that are already present in the reader

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ktkrg** [comment]
> Can you make a list of all the important fixes that are being checked in as part of this commit?

Updated the description.

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PerformanceAnalyzerController.java [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    private static final String RCA_ENABLED_CONF = "rca_enabled.conf";
+    private static final String LOGGING_ENABLED_CONF = "logging_enabled.conf";
+    private static final Logger LOG = LogManager.getLogger(PerformanceAnalyzerController.class);
+
+    private boolean paEnabled;
+    private boolean rcaEnabled;
+    private boolean loggingEnabled;
+    private boolean paEnabledDefaultValue = false;
+    private boolean rcaEnabledDefaultValue = false;
+    private boolean loggingEnabledDefaultValue = false;
```
Comment (yojs) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PerformanceAnalyzerController.java]: Souldn't default values be constants ?

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PerformanceAnalyzerController.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        this.scheduledMetricCollectorsExecutor = scheduledMetricCollectorsExecutor;
+        initPerformanceAnalyzerStateFromConf();
+        initRcaStateFromConf();
+        initLoggingStateFromConf();
+    }
+
+    /**
+     * Returns the current state of performance analyzer.
+     * <p>
+     * This setting is indicative of both the engine and the plugin state.
```
Comment (yojs) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PerformanceAnalyzerController.java]: we are using agent and engine interchangeably. We might want to stick to one ?

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PerformanceAnalyzerController.java [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+     * When enabled, both the writer and the engine are active. The writer captures metrics and stores it
```
Comment (yojs) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PerformanceAnalyzerController.java]: And here writer and the plugin :)

**Pull request merged**